src/USER-RANN/pair_rann.cpp

```diff
@@ -0,0 +1,1261 @@
+      fingerprints[i][j]->allocate();
+    }
+  }
+  allocated=1;
+}
+
+void PairRANN::read_file(char *filename)
+{
+  FILE *fp;
+  int eof = 0,i,j,k,l;
+  int n,nwords;
+  std::string line,line1;
+  int longline = 4096;
+  char linetemp[longline];
+  char *ptr;
+  bool comment;
+  char str[128];
+  std::vector<std::string> linev,line1v;
+  fp = utils::open_potential(filename,lmp,nullptr);
+  if (fp == nullptr) {error->one(FLERR,"Cannot open RANN potential file");}
+  ptr=fgets(linetemp,longline,fp);
+  strip_comments(linetemp,fp);
+  line=linetemp;
+  while (eof == 0) {
+    ptr=fgets(linetemp,longline,fp);
+    if (ptr == NULL) {
+      if (check_potential()) {
+        error->one(FLERR,"Invalid syntax in potential file, values are inconsistent or missing");
+      }
+      else {
+        eof = 1;
+        break;
+      }
+    }
+    strip_comments(linetemp,fp);
+    line1=linetemp;
+    Tokenizer values = Tokenizer(line,": ,\t_\n");
+    Tokenizer values1 = Tokenizer(line1,": ,\t_\n");
+    linev = values.as_vector();
+    line1v = values1.as_vector();
+    if (linev[0].compare("atomtypes")==0)read_atom_types(linev,line1v);
+    else if (linev[0].compare("mass")==0)read_mass(linev,line1v);
+    else if (linev[0].compare("fingerprintsperelement")==0)read_fpe(linev,line1v);
+    else if (linev[0].compare("fingerprints")==0)read_fingerprints(linev,line1v);
+    else if (linev[0].compare("fingerprintconstants")==0)read_fingerprint_constants(linev,line1v);
+    else if (linev[0].compare("networklayers")==0)read_network_layers(linev,line1v);
+    else if (linev[0].compare("layersize")==0)read_layer_size(linev,line1v);
+    else if (linev[0].compare("weight")==0)read_weight(linev,line1v,fp);
+    else if (linev[0].compare("bias")==0)read_bias(linev,line1v,fp);
+    else if (linev[0].compare("activationfunctions")==0)read_activation_functions(linev,line1v);
+    else if (linev[0].compare("calibrationparameters")==0)continue;//information on how the network was trained
+    else if (linev[0].compare("screening")==0)read_screening(linev,line1v);
+    else error->one(FLERR,"Could not understand file syntax: unknown keyword");
+    ptr=fgets(linetemp,longline,fp);
+    strip_comments(linetemp,fp);
+    if (ptr == NULL) {
+      if (check_potential()) {
+        error->one(FLERR,"Invalid syntax in potential file, values are inconsistent or missing");
+      }
+      else {
+        eof = 1;
+        break;
+      }
+    }
+    line=linetemp;
+  }
+}
+
+void PairRANN::strip_comments(char * line,FILE* fp) {
+  char *ptr;
+  bool comment;
+  utils::trim_comment(line);
+  if (strlen(line)==0) {
+    comment = true;
+    while (comment==true) {
+    ptr = fgets(line,4096,fp);
+    if (ptr==NULL) error->one(FLERR,"Unexpected end of RANN file");
+    utils::trim_comment(line);
+    if (strlen(line) == 0) continue;
+    comment = false;
+    }
+  }
+}
+
+void PairRANN::read_atom_types(std::vector<std::string> line,std::vector<std::string> line1) {
```

Comment:
why make this a function? this would be simpler (and better to follow) to just do in place. BTW: appending an item to a vector is simply done with `vector::push_back()`.

Reply:
It is easier to read the potential file parser if all the pieces of the file are read in a consistent way. Thus, each section of the file has its own function and the read_file function can have a nice compact block of conditionals to check where to direct each pair of lines for parsing. This one is pretty short, but it is better to leave it alone. vector::push_back() has now been implemented.